src/threadWorks.hpp

```diff
@@ -1,26 +1,19 @@
 #pragma once
 
-#include <iostream>
 #include <atomic>
 #include <mutex>
 #include <string>
-#include <chrono>
-#include <thread>
-#include <vector>
-#include <cctype>
-#include <csignal>
-#include "matchBuilder.hpp"
-#include "utilities.hpp"
-#include "apiClient.hpp"
-#include "oceanBuilder.hpp"
+
+class apiClient;
+class MillisecondClock;
```

Comment:
I was talking about simply declaring the type when possible instead of including the whole header file. This file made a great case for this, since you only use references to objects of these types in `apiClient&` and `MillisecondClock&`. Since those are types with no need for any implementation details in this file, the type declaration is sufficient. This avoids bringing in the whole headers for those types which include other dependencies which bloat the compilation units that only need `threadWorks.hpp` without `apiClient.hpp` or `MillisecondClock.hpp`.

Reply:
This seems like a "great to know" type of thing with C++. Im still a little confused on this part but ill dig into it